langlib/lang.regexp/src/main/ballerina/regexp.bal

```diff
@@ -174,30 +175,30 @@ public function replace(RegExp reExp, string str, Replacement replacement, int s
     } else {
         replacementStr = replacement;
     }
-    return replaceFromString(reExp, str, replacementStr, startIndex);
+    return replaceFromString('re, str, replacementStr, startIndex);
 }
 
-function replaceFromString(RegExp reExp, string str, string replacementStr, int startIndex) returns string = @java:Method {
+isolated function replaceFromString(RegExp reExp, string str, string replacementStr, int startIndex) returns string = @java:Method {
     'class: "org.ballerinalang.langlib.regexp.Replace",
     name: "replaceFromString"
 } external;
 
 # Replaces all occurrences of a regular expression.
-public function replaceAll(RegExp reExp, string str, Replacement replacement, int startIndex = 0) returns string {
-    Groups? findResult = findGroups(reExp, str, startIndex);
+public isolated function replaceAll(RegExp 're, string str, Replacement replacement, int startIndex = 0) returns string {
```

Comment:
Diff for not using `@isolatedParam` in the parameter depends on https://github.com/ballerina-platform/ballerina-spec/issues/1130#issuecomment-1275589846